### PR TITLE
Soybean parameterization

### PR DESCRIPTION
**Pull request by uphone87**
This PR is mainly to incorporate a couple of changes I made on parameterizing soybean. This mainly includes a few scaling factors on enzymes and some bugs fixed. The points below may be helpful for a general view,

- This version has merged the latest changes from the Master. 
- This version is stable and consistency with the Matlab version.
- This version has been used in the coupled BioCro-ePhotosynthesis.
- More details on the parameterization have been included in the manuscript.

### CODE REVIEW

File: run/compile_test.sh
Comment (astro-friedel) [inline run/compile_test.sh:1]: I would strongly recommend not using a script to build the tests. You should add them to CMake (I can provide some pointers if needed). The issues I see with the script are:

1. The paths are hard coded, and thus will only work on your machine
2. g++ is forces to be the compiler, but not everyone has or uses the gcc suite (clang is also quite popular)

If you convert this to CMake:

1. All of the paths can be specified as relative to the build or as variables like Boost_INCLUDE_DIRS
2. CMake will determine what compiler to use for the given system.

File: run/run_ephoto_test.cpp
Comment (astro-friedel) [inline run/run_ephoto_test.cpp:1]: Is this really a test? It never runs any comparisons, or it it just to test that things don't crash?

File: CMakeLists.txt
Comment (astro-friedel) [inline CMakeLists.txt:1]: Is there a particular reason why you changed the linking from boost_regex to Boost_LIBRARIES? I don't see any additions new boost code that would need this.

**Review by astro-friedel — CHANGES_REQUESTED** [review]
I am assuming that the changing of the variables will not break this code for others who may be using it for things other than soybeans.

**Comment by uphone87** [comment]
> I am assuming that the changing of the variables will not break this code for others who may be using it for things other than soybeans.

Yes. I think it would not affect other crops.

File: run/compile_test.sh
Comment (uphone87) [reply run/compile_test.sh:1]: I have removed the run folder, which is unnecessary to be merged to master.

File: run/run_ephoto_test.cpp
Comment (uphone87) [reply run/run_ephoto_test.cpp:1]: Removed the **run** folder

File: CMakeLists.txt
Comment (uphone87) [reply CMakeLists.txt:1]: In a test we tried on BioCluster, there was an issue with **boost_regex**. It seems that the dynamic library was not able to find the path to **boost_regex** automatically. Using **Boost_LIBRARIES** fixed the problem.

Comment (astro-friedel) [reply CMakeLists.txt:1]: Interesting, they must have a custom (or very old) installation of Boost on their system.

**Review by astro-friedel — APPROVED** [review]
We may want to look into exposing more of these variables that keep changing as either command line options or settable from a config file. That way you would not need to recompile after each change.

**Pull request merged**